src/Code02.DaylightTimeZone.pas

```diff
@@ -36,32 +39,209 @@ interface
 
   Wymagane jest użycie `TMyHttpGet.GetWebsiteContent` do pobrania zawartości strony
   Przykład wywołania:
-    aHtmlPageContent := TMyHttpGet.GetWebsiteContent(‘http://delphi.pl/’);
+  aHtmlPageContent := TMyHttpGet.GetWebsiteContent(‘http://delphi.pl/’);
 }
 
-function IsDaylightSaving(const area: string; year: word): boolean;
-function GetDaylightStart(const area: string; year: word): TDateTime;
-function GetDaylightEnd(const area: string; year: word): TDateTime;
+function IsDaylightSaving(const area: string; year: Word): boolean;
+function GetDaylightStart(const area: string; year: Word): TDateTime;
+function GetDaylightEnd(const area: string; year: Word): TDateTime;
+function MakeValidUrl(const area: string; year: Word): string;
+function GetPageContent(const area: string; year: Word): string;
+procedure SaveDataToRecords(const area: string; year: Word);
+function DataInRecords(const data: string): TJSONObject;
+function GetDaylightFromRecord(const area: string; year: Word): boolean;
+function JsonFromDB: TJSONObject;
+function PageStringToDateTime(const date: string; year: Word): TDateTime;
+function GetDaylightDateFromRecord(const area: string; year: Word; param: string): TDateTime;
 
 implementation
 
 uses
   System.SysUtils,
+  System.IOUtils,
+  System.DateUtils,
+  System.StrUtils,
   Code02.HttpGet;
 
+const
+  cUrlStr = 'https://www.timeanddate.com/time/change/%s?year=%d';
+  cNotDTSdiv = '<div class="alert warning">';
+  cDBfile = 'DBrecords.json';
+  cJsonDTS = 'DTS';
+  cJsonStart = 'start';
+  cJsonEnd = 'end';
+
+function GetDaylightDateFromRecord(const area: string; year: Word; param: string): TDateTime;
```

Comment:
Lepiej użyć typu wyliczeniowego (enum) zamiast `param: string`

Reply:
Zastanawiałem się nad tym ale później musiałbym i tak dekodować to na stringa, żeby wyciągnąć dane. Chyba, że coś mi umknęło.